<script>


    const getData = async () => {
        const response = await fetch('https://jsonplaceholder.typicode.com/todos/');

        if(response.status !== 200){
            throw new Error('Can not get data!')
        }
        const data = await  response.json();
        return data;
    };

    getData()
        .then(data=>console.log(data[0]))
        .catch(err=>console.log(err.message));


    {/*


    //Part One


    const getTodos = (callback) => {
        const request = new XMLHttpRequest();

        request.addEventListener('readystatechange', ()=>{
            if(request.readyState === 4 && request.status === 200){
                callback(undefined, request.responseText);
            } else  if(request.readyState === 4){
                callback('Could not fetch data', undefined);
            }
        })

        request.open('GET', 'https://jsonplaceholder.typicode.com/todos/');
        request.send();

    };

    console.log(1);
    console.log(2);
    getTodos((err, data)=>{
        console.log('callback fired')
        if(err){
            console.log(err);
        }else {
            console.log(data);
        }
    });

    console.log(3);
    console.log(4);


    //Part Two


    const getData = (resource, callback) => {
        const request = new XMLHttpRequest();

        request.addEventListener('readystatechange', ()=>{
            //console.log(request, request.readyState);

            if (request.status===200 && request.readyState === 4){
                const data = JSON.parse(request.responseText);
                callback(undefined, data);
            } else if (request.readyState === 4){
                callback('could not fetch data', undefined);
            }
        });

        request.open('GET', resource);
        request.send();

    }

    getData('parents.json',(err, data)=>{
        console.log(data[0].name)
        getData('children.json', (err, data)=>{
            console.log(data[0].name);
        })
    });


    //Part Three

        const getData = () => {
        return new Promise((resolve, reject)=>{
            resolve('Some data');
            //reject('Some error');
        })
    }


    getData().then((data)=>{
        console.log(data)
    }).catch((error)=>{
        console.log(error)
    });


    //Part Four Promise

    const getSomeData = (resource) => {
        return new Promise((resolve, reject)=>{
            const request = new XMLHttpRequest();

            request.addEventListener('readystatechange', ()=>{
                //console.log(request, request.readyState);

                if (request.status===200 && request.readyState === 4){
                    const data = JSON.parse(request.responseText);
                    resolve(data);
                } else if (request.readyState === 4){
                    reject('Sorry, could not get the data!');
                }
            });

            request.open('GET', resource);
            request.send();
        });
    }

    getSomeData('children.json').then((data)=>{
        console.log(data[0]);
    }).catch((err)=>{
        console.log(err);
    })


    //Part Five Promise Chaining

 const getSomeData = (resource) => {
        return new Promise((resolve, reject)=>{
            const request = new XMLHttpRequest();

            request.addEventListener('readystatechange', ()=>{
                //console.log(request, request.readyState);

                if (request.status===200 && request.readyState === 4){
                    const data = JSON.parse(request.responseText);
                    resolve(data);
                } else if (request.readyState === 4){
                    reject('Sorry, could not get the data!');
                }
            });

            request.open('GET', resource);
            request.send();
        });
    }

    getSomeData('children.json').then( data=>{
        console.log('Promise one resolved!',data[0]);
        return getSomeData('parents.json');
    }).then(data=>{
        console.log('Promise two resolved!', data[0]);
        return getSomeData('children.json');
    }).then(data=>{
        console.log('Promise three resolved!', data[1]);
    }).catch((err)=>{
        console.log(err);
    });


    //Part Six The Fetch API

    fetch('https://jsonplaceholder.typicode.com/todos/1').then((response)=>{
        console.log('resolved', response);
        return response.json();
    }).then(data=>{
        console.log(data);
    }).catch((error)=>{
        console.log('rejected', error);
    });



    //Part Seven Async-Await

      const getData = async () => {
        const response = await fetch('https://jsonplaceholder.typicode.com/todos/');

        if(response.status !== 200){
            throw new Error('Can not get data!')
        }
        const data = await  response.json();
        return data;
    };

    getData()
        .then(data=>console.log(data[0]))
        .catch(err=>console.log(err.message));

   */}
</script>